homeassistant/components/slack/notify.py

```diff
@@ -112,6 +151,51 @@ async def _async_send_local_file_message(self, path, targets, message, title):
         except SlackApiError as err:
             _LOGGER.error("Error while uploading file-based message: %s", err)
 
+    async def _async_send_remote_file_message(
+        self, url, targets, message, title, *, username=None, password=None
+    ):
+        """Upload a remote file (with message) to Slack.
+
+        Note that we bypass the python-slackclient WebClient and use aiohttp directly,
```

Comment:
I don't think we should make these kinds of exceptions to our standards. If new functionality is needed that can't go in the existing 3rd party library a new 3rd party library should be made, that wraps the existing library, where the new functionality can go.